mpisppy/opt/sc.py

```diff
@@ -0,0 +1,121 @@
+        if res is not None:
+            raise RuntimeError(f'Found multiple active objectives on {m}')
+        res = obj
+    return res
+
+
+class _SCInterface(parapint.interfaces.MPIStochasticSchurComplementInteriorPointInterface):
+    def __init__(self,
+                 local_scenario_models: Dict[str, _BlockData],
+                 all_scenario_names: List[str],
+                 comm: MPI.Comm,
+                 ownership_map: Dict):
+        self.local_scenario_models = local_scenario_models
+
+        models = list(local_scenario_models.values())
+        ref_model = models[0]
+        models = models[1:]
+        self.nonant_vars = list()
+
+        for node in ref_model._mpisppy_node_list:
+            ndn = node.name
+            for ndx, v in enumerate(node.nonant_vardata_list):
+                self.nonant_vars.append((ndn, ndx))
+
+        super(_SCInterface, self).__init__(scenarios=all_scenario_names,
+                                           nonanticipative_var_identifiers=self.nonant_vars,
+                                           comm=comm,
+                                           ownership_map=ownership_map)
+
+    def build_model_for_scenario(self,
+                                 scenario_identifier: str) -> Tuple[_BlockData, Dict[Any, _GeneralVarData]]:
+        m = self.local_scenario_models[scenario_identifier]
+
+        _assert_continuous(m)
+
+        active_obj = _get_active_objective(m)
+        active_obj.deactivate()
+        m._mpispypy_weighted_obj = pyo.Objective(expr=m._mpisppy_probability * active_obj.expr, sense=active_obj.sense)
+
+        nonant_vars = dict()
+        for node in m._mpisppy_node_list:
+            ndn = node.name
+            for ndx, v in enumerate(node.nonant_vardata_list):
+                nonant_vars[ndn, ndx] = v
+
+        if len(nonant_vars) != len(self.nonant_vars):
+            raise ValueError(f'Number of non-anticipative variables is not consistent in scenario {scenario_identifier}.')
+
+        return m, nonant_vars
+
+
+class SchurComplement(SPBase):
+    def __init__(self,
+                 options: Union[Dict, SCOptions],
+                 all_scenario_names: List,
+                 scenario_creator: Callable,
+                 scenario_creator_kwargs: Optional[Dict] = None,
+                 all_nodenames=None,
+                 mpicomm=None,
+                 model_name=None,
+                 suppress_warnings=False):
+        super(SchurComplement, self).__init__(options=options,
+                                              all_scenario_names=all_scenario_names,
+                                              scenario_creator=scenario_creator,
+                                              scenario_creator_kwargs=scenario_creator_kwargs,
+                                              all_nodenames=all_nodenames,
+                                              mpicomm=mpicomm)
+
```

Comment:
We should probably check two flags set by `SPBase.__init__`: `self.bundling` (I think should be `False`) and `self.multistage` (probably also should be `False`).

Reply:
I was actually hoping this would work for multistage problems, but I have not tested it yet...